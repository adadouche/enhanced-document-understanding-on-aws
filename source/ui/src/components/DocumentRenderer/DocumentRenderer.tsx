/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import 'react-pdf/dist/esm/Page/TextLayer.css';
import './DocumentRenderer.css';

import { Marks, TextractTable } from '../../utils/interfaces';
import PDF, { DocumentMarks } from '../Pdf/Pdf';

type DocumentRendererProps = {
    selectedDocumentFileType: string | null;
    selectedDocumentUrl: string | null;
    currentPageNumber?: number;
    switchPage?: Function;
    marks?: Marks[];
    tables?: TextractTable[];
    previewRedaction?: string;
    retrieveSignedUrl: Function;
};

export default function DocumentRenderer(props: DocumentRendererProps) {
    return (
        <div>
            {props.selectedDocumentFileType === 'pdf' &&
                props.selectedDocumentUrl &&
                props.currentPageNumber &&
                props.switchPage && (
                    <PDF
                        pdfUrl={props.selectedDocumentUrl}
                        currentPageNumber={props.currentPageNumber}
                        switchPage={props.switchPage}
                        marks={props.marks}
                        tables={props.tables}
                        previewRedaction={props.previewRedaction}
                        retrieveSignedUrl={props.retrieveSignedUrl}
                    />
                )}
            {(props.selectedDocumentFileType === 'jpg' ||
                props.selectedDocumentFileType === 'png' ||
                props.selectedDocumentFileType === 'jpeg') &&
                props.selectedDocumentUrl && (
                    <div className="pdf-viewer">
                        <DocumentMarks
                            marks={props.marks}
                            tables={props.tables}
                            previewRedaction={props.previewRedaction}
                        >
                            <img src={props.selectedDocumentUrl} className="image" alt="file" data-testid="image" />
                        </DocumentMarks>
                    </div>
                )}
            {props.selectedDocumentFileType !== 'pdf' &&
                props.selectedDocumentFileType !== 'jpg' &&
                props.selectedDocumentFileType !== 'jpeg' &&
                props.selectedDocumentFileType !== 'png' && <div>Invalid file type</div>}
        </div>
    );
}
