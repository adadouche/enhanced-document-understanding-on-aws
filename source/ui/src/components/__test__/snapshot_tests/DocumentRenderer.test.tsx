/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import '@testing-library/jest-dom';
import { marks1, tables } from '../../../__test__/test_data';
import DocumentRenderer from '../../DocumentRenderer/DocumentRenderer';
import renderer from 'react-test-renderer';

jest.mock('@cloudscape-design/components', () => {
    const Components = jest.genMockFromModule('@cloudscape-design/components') as any;
    for (const componentName of Object.keys(Components)) {
        Components[componentName] = componentName;
    }
    return Components;
});

jest.mock('react-pdf', () => ({
    pdfjs: { GlobalWorkerOptions: { workerSrc: 'abc' } },
    Document: ({ onLoadSuccess = (pdf = { numPages: 4 }) => pdf.numPages }) => {
        return <div>{onLoadSuccess({ numPages: 4 })}</div>;
    },
    Outline: null,
    Page: () => <div>def</div>
}));

describe('DocumentRenderer component', () => {
    it('DocumentRenderer snapshot test', async () => {
        const selectedDocumentFileType = 'pdf';
        const selectedDocumentUrl = 'fake-url';
        const currentPageNumber = 1;
        const switchPage = jest.fn();
        const tree = renderer
            .create(
                <DocumentRenderer
                    selectedDocumentFileType={selectedDocumentFileType}
                    selectedDocumentUrl={selectedDocumentUrl}
                    currentPageNumber={currentPageNumber}
                    switchPage={switchPage}
                    marks={marks1}
                    tables={tables}
                    retrieveSignedUrl={jest.fn()}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
