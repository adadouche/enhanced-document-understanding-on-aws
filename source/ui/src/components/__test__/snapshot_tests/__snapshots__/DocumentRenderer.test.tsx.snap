// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentRenderer component DocumentRenderer snapshot test 1`] = `
<div>
  <div
    className="pdf-viewer"
    data-testid="pdf-box"
  >
    <div
      className="react-pdf__Document document"
    >
      <div
        className="react-pdf__message react-pdf__message--loading"
      >
        Loading PDFâ€¦
      </div>
    </div>
    <Pagination
      ariaLabels={
        Object {
          "nextPageLabel": "Next page",
          "pageLabel": [Function],
          "previousPageLabel": "Previous page",
        }
      }
      currentPageIndex={1}
      data-testid="pdf-pagination"
      onChange={[Function]}
      pagesCount={0}
    />
  </div>
</div>
`;
